libsplinter/protos/service.proto

```diff
@@ -0,0 +1,126 @@
+// Copyright 2018-2020 Cargill Incorporated
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+message ServiceMessage {
+    // The message type for the payload
+    ServiceMessageType message_type = 1;
+
+    // either a message defined below or another message envelope
+    bytes payload = 2;
+}
+
+enum ServiceMessageType {
+    UNSET_SERVICE_MESSAGE_TYPE = 0;
+
+    // Registration-related messages
+    SM_SERVICE_CONNECT_REQUEST = 100;
+    SM_SERVICE_CONNECT_RESPONSE = 101;
+    SM_SERVICE_DISCONNECT_REQUEST = 102;
+    SM_SERVICE_DISCONNECT_RESPONSE = 103;
+
+    // Opaque message contents, with routing information.
+    SM_SERVICE_PROCESSOR_MESSAGE = 200;
+}
+
+message SMConnectRequest {
+    // the name of the circuit the message is meant for
+    string circuit = 1;
+
+    // the unique id of the service that is connecting to the circuit
```

Comment:
Not consistent with the capitalization guidelines for Splinter API/rustdoc comments.  For example, start a comment with a capital letter, and write ID and IDs. 

This comment applies to the other files in this PR.

Reply:
Fixed